Backend/Interview.Domain/Rooms/RoomReviews/Services/RoomReviewService.cs

```diff
@@ -75,57 +73,118 @@ public Task<IPagedList<RoomReviewPageDetail>> FindPageAsync(
             cancellationToken);
     }
 
-    public async Task<RoomReviewDetail> CreateAsync(
-        RoomReviewCreateRequest request, Guid userId, CancellationToken cancellationToken = default)
+    public async Task<RoomReviewDetail> CreateAsync(RoomReviewCreateRequest request, Guid userId, CancellationToken cancellationToken = default)
     {
-        var user = await _userRepository.FindByIdAsync(userId, cancellationToken);
+        var roomParticipant = await _roomParticipantRepository.FindByRoomIdAndUserIdDetailedAsync(request.RoomId, userId, cancellationToken);
+
+        if (roomParticipant is null)
+        {
+            throw new UserException("You are not a member of the room");
+        }
 
-        if (user is null)
+        if (roomParticipant.Room.Status != SERoomStatus.Review)
         {
-            throw NotFoundException.Create<User>(userId);
+            throw new UserException("The room is not in the review status");
         }
 
-        var room = await _roomRepository.FindByIdAsync(request.RoomId, cancellationToken);
+        var roomReview = new RoomReview(roomParticipant) { Review = request.Review, };
+
+        await _roomReviewRepository.CreateAsync(roomReview, cancellationToken);
+
+        return RoomReviewDetailMapper.Instance.Map(roomReview);
+    }
+
+    public async Task<UpsertReviewResponse> UpsertAsync(RoomReviewCreateRequest request, Guid userId, CancellationToken cancellationToken = default)
+    {
+        var transaction = await _db.Database.BeginTransactionAsync(IsolationLevel.ReadCommitted, cancellationToken);
```

Comment:
add `using` keyword

Reply:
o, my god. I really forgot it's keyword.